redundant-bmc/src/rbmc_tool.cpp

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "services_impl.hpp"
+
+#include <CLI/CLI.hpp>
+#include <xyz/openbmc_project/ObjectMapper/client.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/Sibling/client.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/client.hpp>
+#include <xyz/openbmc_project/State/BMC/client.hpp>
+
+#include <format>
+#include <iostream>
+
+using Sibling =
+    sdbusplus::client::xyz::openbmc_project::state::bmc::redundancy::Sibling<>;
+using Redundancy =
+    sdbusplus::client::xyz::openbmc_project::state::bmc::Redundancy<>;
+using BMCState = sdbusplus::client::xyz::openbmc_project::state::BMC<>;
+using Role = Redundancy::Role;
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<std::string> getBMCState(sdbusplus::async::context& ctx)
+{
+    using ObjectMapper =
+        sdbusplus::client::xyz::openbmc_project::ObjectMapper<>;
+    using StateMgr = sdbusplus::client::xyz::openbmc_project::state::BMC<>;
+
+    try
+    {
+        auto mapper = ObjectMapper(ctx)
+                          .service(ObjectMapper::default_service)
+                          .path(ObjectMapper::instance_path);
+
+        std::string statePath = std::string{BMCState::namespace_path::value} +
+                                '/' + BMCState::namespace_path::bmc;
+        std::vector<std::string> stateIface{BMCState::interface};
+
+        auto object = co_await mapper.get_object(statePath, stateIface);
+        auto service = object.begin()->first;
+
+        auto stateMgr = StateMgr(ctx).service(service).path(statePath);
+        auto bmcState = co_await stateMgr.current_bmc_state();
+
+        auto stateString = BMCState::convertBMCStateToString(bmcState);
+        co_return stateString.substr(stateString.find_last_of('.') + 1);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        // Just show the error in the state field
+        co_return e.what();
+    }
+}
+
+// NOLINTBEGIN
+sdbusplus::async::task<>
+    displayLocalBMCInfo(sdbusplus::async::context& ctx, bool extended)
+// NOLINTEND
+{
+    auto redundancy = sdbusplus::async::proxy()
+                          .service(Redundancy::interface)
+                          .path(Redundancy::instance_path)
+                          .interface(Redundancy::interface);
+
+    std::cout << "Local BMC\n";
+    std::cout << "-----------------------------\n";
+
+    try
+    {
+        auto props =
+            co_await redundancy
+                .get_all_properties<Redundancy::PropertiesVariant>(ctx);
+
+        auto role =
+            Redundancy::convertRoleToString(std::get<Role>(props.at("Role")));
+        role = role.substr(role.find_last_of('.') + 1);
```

Comment:
Could you add a comment on what we're getting back to help explain the "find_last_of(.)"?

Reply:
done